src/ert/_c_wrappers/enkf/lark_parser.py

```diff
@@ -450,22 +455,32 @@ def handle_includes(tree: Tree, config_file: str):
             continue  # This is either a newline or a comment
         if node.children[0].data == "include":
             inc_node = node.children[0]
+
             if len(inc_node.children) > 1:
                 raise ConfigValidationError(
                     "Keyword:INCLUDE must have exactly one argument",
                     config_file=config_file,
                 )
-            val = inc_node.children[0].children[0]
-            if not isinstance(val, FileContextToken):
+            file_to_include = inc_node.children[0].children[0]
+            if not isinstance(file_to_include, FileContextToken):
                 raise ConfigValidationError(
-                    f"INCLUDE keyword must be given filepath, got {val!r}",
+                    f"INCLUDE keyword must be given filepath, got {file_to_include!r}",
                     config_file=config_file,
                 )
-            if not os.path.isabs(val):
-                val = os.path.normpath(os.path.join(config_dir, val))
+            if not os.path.isabs(file_to_include):
+                file_to_include = os.path.normpath(
+                    os.path.join(config_dir, file_to_include)
+                )
 
-            sub_tree = _parse_file(val, "INCLUDE")
-            handle_includes(sub_tree, val)
+            sub_tree = _parse_file(file_to_include, "INCLUDE")
+            if file_to_include in already_included_files:
```

Comment:
Doesn't it make sense to parse the file only if it isn't a cyclical import?

Reply:
Oops! Fixed now